src/Featurizers/SimpleRollingWindowFeaturizer.h

```diff
@@ -332,16 +323,33 @@ void SimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::complete_traini
 // |
 // ----------------------------------------------------------------------
 template <typename InputT, size_t MaxNumTrainingItemsV>
-GrainedSimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::GrainedSimpleRollingWindowEstimator(AnnotationMapsPtr pAllColumnAnnotations, SimpleRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t maxWindowSize, std::uint32_t minWindowCount) :
+GrainedSimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::GrainedSimpleRollingWindowEstimator(AnnotationMapsPtr pAllColumnAnnotations, SimpleRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t maxWindowSize, std::uint32_t minWindowSize) :
     BaseType(
         GrainedSimpleRollingWindowEstimatorName,
         pAllColumnAnnotations,
-        [pAllColumnAnnotations, maxWindowSize, windowCalculation, horizon, minWindowCount] () {
+        [pAllColumnAnnotations, maxWindowSize, windowCalculation, horizon, minWindowSize] () {
             return Components::GrainEstimatorImpl<GrainType, SimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>> (
                 pAllColumnAnnotations,
-                [maxWindowSize, windowCalculation, horizon, minWindowCount](AnnotationMapsPtr pAllColumnAnnotationsParam) {
-                    return SimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>(std::move(pAllColumnAnnotationsParam), windowCalculation, horizon, maxWindowSize, minWindowCount);
-                }
+                [windowCalculation, horizon, maxWindowSize, minWindowSize](AnnotationMapsPtr pAllColumnAnnotationsParam) {
+                    return SimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>(
+                        std::move(pAllColumnAnnotationsParam),
+                        windowCalculation,
+                        horizon,
+                        maxWindowSize,
+                        minWindowSize
+                    );
+                },
+                [windowCalculation, horizon, maxWindowSize, minWindowSize](GrainType const &) {
+                    return typename SimpleRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::TransformerUniquePtr(
+                        new SimpleRollingWindowTransformer<InputT, MaxNumTrainingItemsV>(
+                            windowCalculation,
+                            horizon,
+                            maxWindowSize,
+                            minWindowSize
+                        )
+                    );
+                },

```

Comment:
If you are already passing in the functor to directly create the transformer and its marked as a trainingOnlyEstimator, do you even need to use the estimator directly or can you just bypass it?

Reply:
I had the same thought, and unfortunately the answer is that you do need the estimator creation func. This is because the GrainEstimatorImpl creates an instance of the Estimator in order to get its Name when creating the grained estimator name.